crates/sui-framework/tests/collectible_tests.move

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+/// Initializes a simple collection.
+module sui::boars {
+    use sui::tx_context::{TxContext, sender};
+    use sui::transfer::transfer;
+    use sui::collectible;
+    use sui::display;
+    use std::string::utf8;
+    use std::option;
+
+    /// Limit how many objects can be created within this collection.
+    const CAP: u64 = 1000;
+
+    /// An OTW to use when creating a Publisher and CollectionCreatorCap
+    struct BOARS has drop {}
+
+    /// The type of the Collectible; used only as a phantom parameter, and
+    /// does not require abilities. The type use is authorized with the Publisher
+    /// object and is hard-capped to the module (rather than the package).
+    struct Boar has store {}
+
+    fun init(otw: BOARS, ctx: &mut TxContext) {
+        let (pub, display, creator_cap) = collectible::create_collection<BOARS, Boar>(
+            otw, option::some(CAP), ctx
+        );
+
+        display::add_multiple(&pub, &mut display, vector[
```

Comment:
When I try to publish this module, I got an MoveAbort within this function with error code `ENotOwner`, is this expected?

Reply:
My guess is that the `T` in `is_package<T>(pub)` is `Collectible<Boar>` but the publisher object is created for `Boar`.

```
/// Sets multiple `fields` with `values`.
    entry public fun add_multiple<T: key>(
        pub: &Publisher, d: &mut Display<T>, fields: vector<String>, values: vector<String>
    ) {
        let len = vector::length(&fields);
        assert!(is_package<T>(pub), ENotOwner);
        assert!(len == vector::length(&values), EVecLengthMismatch);

        let i = 0;
        while (i < 0) {
            add_internal(d, *vector::borrow(&fields, i), *vector::borrow(&values, i));
            i = i + 1;
        };
    }
```

Reply:
Oh, right, sorry. I’ll send a patch shortly.

Reply:
Replied in the other PR. But that's definitely not the expected behavior. It's hard to test within Sui Framework as all created modules get created in the same package (therefore we can't cover case with different packages).

Addressed in #8659 - removes access control checks in all of the functions except `display::new()` moving the responsibility to protect and control use of Display to the implementors.